### PR TITLE
Overload `fast_plain_mtext` to different return types depending on `split=`

### PR DESCRIPTION
**Pull request by bswck-hai**
- [x] Type checked the code.

### CODE REVIEW

**Comment by bswck-hai** [comment]
@mozman Please review.

**Comment by mozman** [comment]
Thanks for your contribution, but currently I integrate only bug fixes for more information see my latest posting #1175.

File: src/ezdxf/tools/text.py [hunk 1]
```diff
@@ -9,11 +9,13 @@
     Iterable,
     Iterator,
     TYPE_CHECKING,
+    Literal,
     Union,
     Optional,
     Callable,
     NamedTuple,
     Any,
+    overload,
```
Comment (mozman) [inline src/ezdxf/tools/text.py:18]: has to be imported from ` typing_extensions` , not available in Python 3.9 (requires 3.11)

File: src/ezdxf/tools/text.py [hunk 2]
```diff
@@ -414,8 +416,18 @@ def plain_text(text: str) -> str:
...
+
+
+@overload
+def fast_plain_mtext(text: str, split: Literal[False] = False) -> str:
+    ...
+
+
 # pylint: disable-next=too-many-branches
-def fast_plain_mtext(text: str, split=False) -> Union[list[str], str]:
+def fast_plain_mtext(text: str, split: bool = False) -> Union[list[str], str]:
```
Comment (mozman) [inline src/ezdxf/tools/text.py:430]: I don't add `bool` for such cases, type inference by `False` and `True`, shorter signatures are better, therefore return `list[str]|str` .

File: src/ezdxf/tools/text.py (hunk 1 above)
Comment (bswck-hai) [reply src/ezdxf/tools/text.py:18]: I believe you misread the documentation. `typing.overload` is the part of PEP 484.

File: src/ezdxf/tools/text.py (hunk 2 above)
Comment (bswck-hai) [reply src/ezdxf/tools/text.py:430]: I'd argue that shorter signatures aren't better when compared against actually accurate signatures.
Type inference from `False` is `Any` in mypy (not mentioning it erroring in strict mode).

File: src/ezdxf/tools/text.py [hunk 3]
```diff
@@ -9,11 +9,13 @@
     Iterable,
     Iterator,
     TYPE_CHECKING,
+    Literal,
```
[the remark this answers is not in the document]
Comment (bswck-hai) [reply src/ezdxf/tools/text.py:12]: Why? `Literal` has been there since Python 3.8, `ezdxf` supports 3.9+.

File: src/ezdxf/tools/text.py (hunk 2 above)
Comment (bswck-hai) [reply src/ezdxf/tools/text.py:430]: > therefore return `list[str]|str`
Migrating to PEP 604 is a topic for a different PR.

**Comment by bswck-hai** [comment]
@mozman could you please reopen this?

**Pull request closed without merging**